### PR TITLE
model quantization

### PR DESCRIPTION
**Pull request by AndreSlavescu**

### CODE REVIEW

File: docs/onnx-conversion.md [hunk 1]
```diff
@@ -316,6 +316,46 @@ Sparse vector output after thresholding: [[[0.         0.23089279 0.14276895 ...
 
 All of these definitions are modularized in ```run_onnx_inference(model_path, model_name, text, threshold)```.
 
+## Quantization
+
+### Run End to End Quantization
```
Comment (lintool) [inline docs/onnx-conversion.md:321]:
```suggestion
### Run End-to-End Quantization
```

File: docs/onnx-conversion.md [hunk 2]
```diff
@@ -316,6 +316,46 @@ Sparse vector output after thresholding: [[[0.         0.23089279 0.14276895 ...
...
+```
+
+For this example, we will continue with the SPLADE++ Ensemble Distil model.
+
+To run the script for running inference, run the following sequence of commands:
+```bash
+# Begin by going to the appropriate directory
+cd src/main/python/onnx
+# Now run the script
+python3 quantize_onnx_model.py --model_path models/splade-cocondenser-ensembledistil-optimized-8bit.onnx \
```
Comment (lintool) [inline docs/onnx-conversion.md:335]:
```suggestion
python quantize_onnx_model.py --model_path models/splade-cocondenser-ensembledistil-optimized-8bit.onnx \
```

File: docs/onnx-conversion.md [hunk 3]
```diff
@@ -316,6 +316,46 @@ Sparse vector output after thresholding: [[[0.         0.23089279 0.14276895 ...
...
+                                    --model_name naver/splade-cocondenser-ensembledistil
+```
+
+So what actually happens under the hood? The following sections will discuss the key parts of the above script:
+
+### Quantizing the Model to 8-bit
+
+As seen below, the model name and extension are extracted from the presented optimized onnx model file, and a custom name with 8-bit is created. 
+
+In terms of the quantization semantics, only the ```model_input``` and ```model_output``` are needed as specifications to the target model. The other two arguments are needed for specifying the desired weight datatype with ```weight_type=QuantType.QInt8``` as well as the default tensor type ```extra_options={'DefaultTensorType': onnx.TensorProto.FLOAT}```
```
Comment (lintool) [inline docs/onnx-conversion.md:345]: are you sure you want triple backquote? This typesets the contents as a block... vs. just a single backquote?

make sure this is how you want the text to look?

File: docs/onnx-conversion.md (hunk 2 above)
Comment (lintool) [reply docs/onnx-conversion.md:335]: And elsewhere python3 -> python please.

**Review by lintool — CHANGES_REQUESTED** [review]

File: docs/onnx-conversion.md (hunk 3 above)
Comment (AndreSlavescu) [reply docs/onnx-conversion.md:345]: Will fix

**Review by lintool — APPROVED** [review]

**Pull request merged**